.travis.yml

```diff
@@ -33,10 +33,23 @@ before_script:
 
     - make extracheck
 
-script:
-    - make coverage
+os:
+  - linux
+  - osx
+  # - windows -- currently not built for windows
+
+matrix:
+  include:
+    - stage builds
+    script:
+      make build-natives
+
+    # If it builds for all the required platforms, run the unit tests.
+    - stage tests
+    script:
+    # Unit tests will only be run for linux.
+      - if [ "$TRAVIS_OS_NAME" = "linux" ]; then make coverage && make ; fi
```

Comment:
I seem to recall there being some way to make conditionals at the parsing stage, not in the shell command. Have you looked into that? It might save spinning up a whole Mac VM in Travis just to tear it down again immediately.

Reply:
Nice! I've been looking for it, and couldn't find it. So I spun this up to see if it did in fact spawn a new VM for the second build step. I was sort of hoping it wouldn't.